table/block_based_table_reader.cc

```diff
@@ -702,30 +704,27 @@ Status BlockBasedTable::Open(const ImmutableCFOptions& ioptions,
   }
 
   // Read the range del meta block
-  // TODO(wanning&andrewkr): cache range delete tombstone block
   bool found_range_del_block;
-  BlockHandle range_del_handle;
   s = SeekToRangeDelBlock(meta_iter.get(), &found_range_del_block,
-                          &range_del_handle);
+                          &rep->range_del_handle);
   if (!s.ok()) {
     Log(InfoLogLevel::WARN_LEVEL, rep->ioptions.info_log,
         "Error when seeking to range delete tombstones block from file: %s",
         s.ToString().c_str());
```

Comment:
doesn't look guaranteed because it uses BlockHandle::DecodeFrom() (https://github.com/facebook/rocksdb/blob/17f76fc5647ce1a06055c3eae8e5c8055749566f/table/format.cc#L57-L64), which may succeed at populating offset_ and then fail at populating size_ (IsNull() requires both to be zero).